addon/components/polaris-data-table.js

```diff
@@ -0,0 +1,535 @@
+   *
+   * @property footerContent
+   * @type {String|Number|Component}
+   * @public
+   */
+  footerContent: null,
+
+  /**
+   * List of booleans, which maps to whether sorting is enabled or not for each column.
+   * Defaults to false for all columns.
+   *
+   * @property sortable
+   * @type {boolean[]}
+   * @public
+   */
+  sortable: null,
+
+  /**
+   * The direction to sort the table rows on first click or keypress of a sortable column heading.
+   * Defaults to ascending.
+   *
+   * @property defaultSortDirection
+   * @type {String}
+   * @default 'ascending'
+   * @public
+   */
+  defaultSortDirection: 'ascending',
+
+  /**
+   * The index of the heading that the table rows are initially sorted by.
+   * Defaults to the first column.
+   *
+   * @property initialSortColumnIndex
+   * @type {Number}
+   * @default 0
+   * @public
+   */
+  initialSortColumnIndex: 0,
+
+  /**
+   * Callback fired on click or keypress of a sortable column heading.
+   *
+   * @property onSort
+   * @type {function}
+   * @default no-op
+   * @public
+   */
+  onSort(/* headingIndex, direction */) {},
+
+  /**
+   * @property collapsed
+   * @type {boolean}
+   * @default false
+   * @private
+   */
+  collapsed: false,
+
+  /**
+   * @property columnVisibilityData
+   * @type {Object[]}
+   * @private
+   */
+  columnVisibilityData: null,
+
+  /**
+   * @property previousColumn
+   * @type {Object}
+   * @private
+   */
+  previousColumn: null,
+
+  /**
+   * @property currentColumn
+   * @type {Object}
+   * @private
+   */
+  currentColumn: null,
+
+  /**
+   * @property sorted
+   * @type {boolean}
+   * @default false
+   * @private
+   */
+  sorted: false,
+
+  /**
+   * @property sortedColumnIndex
+   * @type {Number}
+   * @private
+   */
+  sortedColumnIndex: null,
+
+  /**
+   * @property sortDirection
+   * @type {String}
+   * @private
+   */
+  sortDirection: null,
+
+  /**
+   * @property heights
+   * @type {Number[]}
+   * @private
+   */
+  heights: null,
+
+  /**
+   * @property preservedScrollPosition
+   * @type {Object}
+   * @private
+   */
+  preservedScrollPosition: null,
+
+  /**
+   * @property previousTruncate
+   * @type {boolean}
+   * @private
+   */
+  previousTruncate: null,
+
+  /**
+   * @property totalsRowHeading
+   * @type {String}
+   * @private
+   */
+  totalsRowHeading: 'Totals',
+
+  /**
+   * @property eventHandlerParams
+   * @type {Array[]}
+   * @private
+   */
+  eventHandlerParams,
+
+  /**
+   * @property dataTable
+   * @type {HTMLElement}
+   * @private
+   */
+  dataTable: volatileElementLookup('.Polaris-DataTable').readOnly(),
+
+  /**
+   * @property scrollContainer
+   * @type {HTMLElement}
+   * @private
+   */
+  scrollContainer: volatileElementLookup('.Polaris-DataTable__ScrollContainer').readOnly(),
+
+  /**
+   * @property table
+   * @type {HTMLElement}
+   * @private
+   */
+  table: volatileElementLookup('.Polaris-DataTable__Table').readOnly(),
```

Comment:
I don't think these element lookups need to be volatile

Reply:
I guess you're right actually 👍